modules/build/src/main/scala/scala/build/Inputs.scala

```diff
@@ -267,7 +267,7 @@ object Inputs {
           case elem: SourceFile => (elem.path / os.up, true, WorkspaceOrigin.SourcePaths)
           case _: Virtual =>
             val dir = homeWorkspace(validElems, directories)
-            (dir, false, WorkspaceOrigin.HomeDir)
+            (dir, true, WorkspaceOrigin.HomeDir)
```

Comment:
We shouldn't need to change this one. These projects already land under directories such as `~/Library/Caches/ScalaCli/virtual-projects/03/project-ad6a817e`, which already include the inputs hash (here, `03ad6a817e`).